rocAL/rocAL/source/pipeline/ring_buffer.cpp

```diff
@@ -351,9 +419,33 @@ void RingBuffer::increment_write_ptr()
     _wait_for_load.notify_all();
 }
 
-void RingBuffer::set_meta_data( ImageNameBatch names, pMetaDataBatch meta_data)
+void RingBuffer::set_meta_data(ImageNameBatch names, pMetaDataBatch meta_data, bool is_segmentation)
 {
-    _last_image_meta_data = std::move(std::make_pair(std::move(names), meta_data));
+    if(meta_data == nullptr)
+        _last_image_meta_data = std::move(std::make_pair(std::move(names), pMetaDataBatch()));
+    else
+    {
+        _last_image_meta_data = std::move(std::make_pair(std::move(names), meta_data));
```

Comment:
Instead of pushing the MetadataBatch, we need to push the MetadataInfoBatch since we need only the related info

Reply:
Not required now